addons/product_configurator/static/src/js/product_configurator_field.js

```diff
@@ -0,0 +1,152 @@
+            }
+        );
+        if(result && result.product_id) {
+            if (this.props.record.data.product_id != result.product_id.id) {
+                await this.props.record.update({
+                    product_id: [result.product_id, result.product_name],
+                });
+                if (result.has_optional_products) {
+                    this._openProductConfigurator();
+                }
+            }
+        } else {
+            if (!result.mode || result.mode === 'configurator') {
+                this._openProductConfigurator();
+            } else {
+                // only triggered when sale_product_matrix is installed.
+                this._openGridConfigurator();
+            }
+        }
+    },
+
+    async _openProductConfigurator(edit=false) {
+        /**
+         *  `product_custom_attribute_value_ids` records are not loaded in the view bc sub templates
+         *  are not loaded in list views. Therefore, we fetch them from the server if the record is
+         *  saved. Else we use the value stored on the line.
+         */
+        const params = await this.configuratorParams(edit)
+        this.dialog.add(ProductConfiguratorDialog, params);
+    },
+
+    async configuratorParams(edit){
+        const customAttributeValues =
+            this.props.record.data.product_custom_attribute_value_ids.records[0]?.isNew ?
+                this.props.record.data.product_custom_attribute_value_ids.records.map(
+                    record => record.data
+                ) :
+                await this.orm.read(
+                    'product.attribute.custom.value',
+                    this.props.record.data.product_custom_attribute_value_ids.currentIds,
+                    ["custom_product_template_attribute_value_id", "custom_value"]
+                );
+        return {
+            productTemplateId: this.props.record.data.product_tmpl_id[0],
+            ptavIds: this.props.record.data.product_template_attribute_value_ids.records.map(
+                record => record.resId
+            ).concat(this.props.record.data.product_no_variant_attribute_value_ids.records.map(
+                record => record.resId
+            )),
+            customAttributeValues: customAttributeValues.map(
+                data => {
+                    return {
+                        ptavId: data.custom_product_template_attribute_value_id[0],
+                        value: data.custom_value,
+                    }
+                }
+            ),
+            edit: edit,
+            options: this.props.options,
+            save: async (mainProduct, optionalProducts) => {
+                await this.applyProduct(this.props.record, mainProduct);
+            },
+            discard: () => {},
+            ...this.additionalParams(),
+        };
+    },
+
+    additionalParams() {
+        if(this.props.options?.hideQty){
+            return {companyId: this.props.record.model.root.data.company_id[0]};
+        }
```

Comment:
You can't just decide not to send the quantity to the server just because you decided to hide it. In `website_sale` for example, we'll soon use this same configurator and we want to be able to hide the quantity but compute the right price.